### PR TITLE
BaSD-MS-11: feature/MS-11/sessions-View-CRUD. 

### PR DESCRIPTION
**Pull request by Micaela-89**
CRUD methods created.

List Sessions
Delete Session
Create Session
Update Session

### CODE REVIEW

**Review by aquilesmariani — CHANGES_REQUESTED** [review]
As per discussed in slack, we should refactor the folders so we have entity-related folders inside `public/views`


- styles
- sessions
- admins
- postulants
- ...

Then, inside every folder, include the markup, styles and functionality (index.html, styles.css, app.js)

Also, if you add `app.use(express.static('public'));` into `index.js` file on line 26, you will be able to access those files through `localhost:3000/views/sessions/sessions-list`

File: package.json [hunk 1]
```diff
@@ -34,4 +34,4 @@
     "eslint-plugin-import": "^2.25.3",
     "nodemon": "^2.0.14"
   }
-}
+}
```
[the remark this answers is not in the document]
Comment (Micaela-89) [reply package.json]: Changes done.
Already commited and push them.

**Comment by Micaela-89** [comment]
CRUD methods created.

- List Sessions
- Delete Session
- Create Session
- Update Session

File: public/views/sessions/js/sessions-form.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    fetch(`${window.location.origin}/api/sessions/${sessionId}`)
+      .then((response) => {
+        if (response.status !== 200 && response.status !== 201) {
+          return response.json().then(({ message }) => {
+            throw new Error(message);
+          });
+        }
+        return response.json();
+      }).then((response) => {
+        console.log(response.data);
```
Comment (aquilesmariani) [inline public/views/sessions/js/sessions-form.js:25]: We should get rid of this console logs

File: public/views/sessions/js/sessions-form.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    if (sessionId) {
+      options.method = 'PUT';
+      url = `${window.location.origin}/api/sessions/${sessionId}`;
+    } else {
+      options.method = 'POST';
+      url = `${window.location.origin}/api/sessions/`;
+    }
+    fetch(url, options)
+    .then((response) => {
+        console.log(response.status)
```
Comment (aquilesmariani) [inline public/views/sessions/js/sessions-form.js:61]: Same with this one.

File: public/views/sessions/js/sessions-list.js [hunk 4]
```diff
@@ -1 +1,51 @@
...
+  window.location.href = `${window.location.origin}/views/sessions/sessions-form.html?id=${id}`;
+};
+const openNewSession = () => {
+  window.location.href = `${window.location.origin}/views/sessions/sessions-form.html`;
+};
+
+let tableContent;
+const deleteSession = (id, index, event) => {
+  event.stopPropagation();
+  fetch(`http://localhost:3000/api/sessions/${id}`, { method: 'DELETE' })
```
Comment (aquilesmariani) [inline public/views/sessions/js/sessions-list.js:12]: Still need to replace localhost for `window.location.origin`

File: public/views/sessions/js/sessions-list.js [hunk 5]
```diff
@@ -1 +1,51 @@
...
+    .then((response) => {
+      if (response.status === 200) {
+        document.getElementById('sessionTable').deleteRow(index);
+      }
+    });
+};
+
+window.onload = () => {
+  tableContent = document.getElementById('sessionTable');
+  fetch('http://localhost:3000/api/sessions/')
```
Comment (aquilesmariani) [inline public/views/sessions/js/sessions-list.js:22]: Same on this one.

**Review by aquilesmariani — CHANGES_REQUESTED** [review]
Just a few things to address and will be ready to go!

File: public/views/sessions/js/sessions-list.js (hunk 4 above)
Comment (Micaela-89) [reply public/views/sessions/js/sessions-list.js]: Done!

**Review by aquilesmariani — APPROVED** [review]
Well done Mica!! 🚀

**Pull request merged**